rclcpp/include/rclcpp/executor.hpp

```diff
@@ -270,7 +270,7 @@ class Executor
                         }));
     }
     // Use the number of subscriptions to allocate memory in the handles
-    size_t number_of_subscriptions = subs.size();
+    unsigned long number_of_subscriptions = subs.size();
```

Comment:
This is not about the availability of `size_t`. This fixes a warning that we are later assigned this variable, `number_of_subscriptions`, on line 275 to `subscriber_count` which is an `unsigned long` and that results in a possible loss of precision. Rather than changing the size of the later storage, I changed the size of this temporary data field. Maybe we should use `size_t` everywhere, but I couldn't remember if there was a reason we did not use it in the structures.